AnkiDroid/src/main/java/com/ichi2/anki/CardBrowser.java

```diff
@@ -170,7 +172,26 @@
 
     private static final int EDIT_CARD = 0;
     private static final int ADD_NOTE = 1;
-    private static final int PREVIEW_CARDS = 2;
+
+    ActivityResultLauncher<Intent> mOnPreviewCardsActivityResult = registerForActivityResult(new ActivityResultContracts.StartActivityForResult(), result -> {
+        Timber.d("onPreviewCardsActivityResult: resultCode=%d", result.getResultCode());
+        if (result.getResultCode() == DeckPicker.RESULT_DB_ERROR) {
+            closeCardBrowser(DeckPicker.RESULT_DB_ERROR);
+        }
+        // Previewing can now perform an "edit", so it can pass on a reloadRequired
+        Intent data = result.getData();
+        boolean hasChanges = data != null && (
+                data.getBooleanExtra("reloadRequired", false)
+                        || data.getBooleanExtra("noteChanged", false)
+        );
+        if (hasChanges) {
```

Comment:
I am really picky about compound conditionals, I like them to be as easy to read as possible - something more like this?

```suggestion
        if ((data != nul)l && 
            (data.getBooleanExtra("reloadRequired", false) || data.getBooleanExtra("noteChanged", false)) {
```

Reply:
I liked to have it in a separate help variable to allow the reader not to read the condition to understand the meaning of the if statement, but I can change it :muscle: :)

Reply:
Fixed!

Reply:
I just prefer to go the other direction - make the conditional so easy to mentally parse that it is not a problem (where possible)